src/components/forms/NetworkDevicesForm/edit/NetworkDevicePanel.tsx

```diff
@@ -0,0 +1,327 @@
+
+    if (isCreatingLocal) {
+      const inheritedNetworkNames = inheritedNetworks.map((item) => item.key);
+      const existingNetworkNames = parentFormik.values.devices.map(
+        (d) => d.name,
+      );
+      const allNetworkNames = [
+        ...existingNetworkNames,
+        ...inheritedNetworkNames,
+      ];
+      const deviceName = deduplicateName("eth", 1, allNetworkNames);
+
+      return {
+        name: deviceName,
+        network: defaultNetworkName,
+        acls: defaultAcls,
+        ipv4: "",
+        ipv6: "",
+      };
+    }
+
+    if (isCreatingOverride) {
+      return {
+        name: inheritedDevice.key,
+        network: inheritedDevice.network?.network || defaultNetworkName,
+        acls: combineAcls(
+          getNetworkAclsByNetworkName(inheritedDevice.network?.name || ""),
+          getDeviceAcls(inheritedDevice.network),
+        ).join(","),
+        ipv4: "",
+        ipv6: "",
+      };
+    }
+
+    const deviceNetworkName = device?.network || defaultNetworkName;
+
+    const combinedAcls = combineAcls(
+      getNetworkAclsByNetworkName(deviceNetworkName),
+      getDeviceAcls(device),
+    ).join(",");
+
+    return {
+      name: device?.name || "",
+      network: deviceNetworkName,
+      acls: combinedAcls,
+      ipv4: device?.["ipv4.address"] || "",
+      ipv6: device?.["ipv6.address"] || "",
+    };
+  };
+
+  const formik = useFormik<NetworkDeviceFormValues>({
+    enableReinitialize: true,
+    initialValues: getInitialValues(),
+    validationSchema,
+    onSubmit: (values) => {
+      const allSelectedAcls = values.acls ? values.acls.split(",") : [];
+      const networkAcls = getNetworkAcls(
+        managedNetworks.find((n) => n.name === values.network),
+      );
+      const userSelectedAcls = allSelectedAcls.filter(
+        (acl) => !networkAcls.includes(acl),
+      );
+
+      const deviceData: FormNetworkDevice = {
+        name: values.name,
+        type: "nic",
+        network: values.network,
+        "security.acls":
+          userSelectedAcls.length > 0 ? userSelectedAcls.join(",") : undefined,
+        "ipv4.address": values.ipv4 || undefined,
+        "ipv6.address": values.ipv6 || undefined,
+      };
+
+      if (isCreatingLocal) {
+        const newDevices = [...parentFormik.values.devices, deviceData];
+        parentFormik.setFieldValue("devices", newDevices);
+      } else {
+        const originalDeviceName = deviceName;
+        const { devices } = parentFormik.values;
+        const existingDeviceIndex = devices.findIndex(
+          (d) => d.name === originalDeviceName,
+        );
+        let updatedDevices: FormDevice[];
+        if (existingDeviceIndex !== -1) {
+          updatedDevices = [...devices];
+          updatedDevices[existingDeviceIndex] = deviceData;
+        } else {
+          updatedDevices = [...devices, deviceData];
+        }
+
+        parentFormik.setFieldValue("devices", updatedDevices);
+      }
```

Comment:
```suggestion
        const originalDeviceName = deviceName;
        const { devices } = parentFormik.values;
        const existingDeviceIndex = devices.findIndex(
          (d) => d.name === originalDeviceName,
        );
        let updatedDevices: FormDevice[];
        if (existingDeviceIndex !== -1) {
          updatedDevices = [...devices];
          updatedDevices[existingDeviceIndex] = deviceData;
        } else {
          updatedDevices = [...devices, deviceData];
        }

        parentFormik.setFieldValue("devices", updatedDevices);
```
I think the creation case generalizes into the other ones.